src/main/scala/com/microsoft/ml/spark/cognitive/TextAnalyticsSDK.scala

```diff
@@ -35,6 +36,7 @@ abstract class TextAnalyticsSDKBase[T](val textAnalyticsOptions: Option[TextAnal
 
   protected lazy val textAnalyticsClient: TextAnalyticsClient =
     new TextAnalyticsClientBuilder()
+      .retryPolicy(new RetryPolicy())

```

Comment:
I think we should use the second c'tor of the `RetryPolicy`. The one that allows to set the retryAfter header. Otherwise, we'll use the default exponential backoff strategy that might fail in some cases, and is probably not very efficient.

Reply:
Fixed!